### DISCUSSION TITLE
Ability to get chart version in HelmRelease from ConfigMap

### DISCUSSION THREAD

**Discussion by jancespivo** [fluxcd/flux2#2381] [Proposals] [answered]
It would be great to add ability to get chart version in HelmRelease from ConfigMap. If the version is changed it will install the new version of the chart.

Now it is possible to get helm values from a ConfigMap and trigger a release if is changed.

We have very similar use-case as it is described in https://fluxcd.io/docs/guides/helmreleases/#refer-to-values-in-configmaps-generated-with-kustomize However we also need to control the chart version and not only the values to complete the control.

There is a workaround already in place we would like to remove:  
HelmRelease triggers helm chart that creates another HelmRelease. Values for the first chart contain also the version of the second chart. There might be kustomize's configMapGenerator instead to generate ConfigMap with the version of the final chart. It is definitely less complex and straightforward way how to do it.

Thx!

**Comment by hiddeco** [comment, marked as answer]
Given the kustomize-controller has the concept of [variable substitution](https://fluxcd.io/docs/components/kustomize/kustomization/#variable-substitution), I am wondering if you could not use this instead to inject the value into the `HelmRelease`.

**Reply by jancespivo** [reply]
I didn't know about the variable substitution. It will simplified the workaround a bit for sure. Thx!

**Reply by julanu** [reply]
Hi! I am also interested in this subject, have you managed an implementation of substiting with variable injection?

**Reply by jancespivo** [reply]
We did PoC successfully, however we found it still quite unreadable and hard to understand solution, so we abandoned it.
